components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/FirefoxAccount.kt

```diff
@@ -105,38 +106,44 @@ class FirefoxAccount internal constructor(
      * @param wantsKeys Fetch keys for end-to-end encryption of data from Mozilla-hosted services
      * @return Deferred<String> that resolves to the flow URL when complete
      */
-    override fun beginOAuthFlow(scopes: Array<String>, wantsKeys: Boolean): Deferred<String> {
-        return scope.async { inner.beginOAuthFlow(scopes, wantsKeys) }
+    override fun beginOAuthFlowAsync(scopes: Array<String>, wantsKeys: Boolean): Deferred<String?> {
+        return scope.async {
+            maybeExceptionalOr(logger, "begin oauth flow", { null }) {
```

Comment:
Hm I remember we wanted to rename this to `handleException`: https://github.com/mozilla-mobile/android-components/pull/2955#discussion_r282654458 which I still find easier to read than `maybeExceptional`...